### ISSUE TITLE
Linux file path error

### ISSUE THREAD

**Issue by RealChiefChen** [HooferDevelops/url-music-discs#22]
Distro: Pop!_OS 22.04 LTS

When trying to play a music disc there seems to be an error causing a pair of quotation marks to be included in the path. Because of this it's creating a directory named a single quotation mark ( " ) and adding another one at the end of the audio file.

Example:
/home/<user>/.minecraft/"/home/<user>/.minecraft/config/urlmusicdiscs/client_downloads/<hash>.raw".webm

I'm not an expert in java or general programing but I believe the error is caused by src/client/java/urlmusicdiscs/AudioHandlerClient.java but I may be wrong

**Comment by Plompi** [comment]
can you elaborate the problem further by providing more details or a video? maybe i can look into it. it also seems like the path contains /home/.minecraft twice which is weird too

**Comment by RealChiefChen** [comment]
Yes sorry about that, I tend to just start jumping head first into explanations instead of taking time to get everyone up to speed. Ok, so whenever I play music discs I get the downloading video message as well as down load competed. However, no audio plays I wondered what was causing it so I started poking around and found a folder in .minecraft named " it contained a chain of folders (im going to just post the path to the last folder) /home/(user)/.minecraft/"/home/(user)/.minecraft/config/urlmusicdiscs/client_downloads Inside of client_downloads is all the videos the mod was supposed to download perfectly playable. Now as you noticed the path I posted has two /home/(user)/.minecraft. Because of this my theory is that when the video is downloading the mod tries to create the file at "/home/(user)/.minecraft/config/urlmusicdiscs/client_downloads/(file)" the quotes are throwing it off however causing it to think that " is a directory and since there is no such folder, it creates one. At that point it starts making all the folders in that path as well. Again sorry for my pitiful explanation earlier, but thank you for answering me!

**Comment by Plompi** [comment]
Thank you very much for the detailed explanation. I might have found the cause of the problem. I suspect calling getAbsolutePath on an already absolute Path leads to weird Behavior. I will do some further digging

**Comment by Plompi** [comment]
are you available for further testing? i have commited an update in which i attempted to fix the issue on my forked repo. You could either build it yourself or i can send you the jar File

**Comment by RealChiefChen** [comment]
Yes I am available for more testing. If you could send me the jar file that would be perfect for me.

**Comment by Plompi** [comment]
i hope thats not needed anymore, i set up a vm with PopOS to further investigate the Problem. I'm now at the point where it works for me on the VM with PopOs. It also should have fixed other Linux Distros (I hope 🙏🏻 )

**Comment by RealChiefChen** [comment]
Thank you! I think the fix should apply to all distros seeing as how it was just some quotes being misinterpreted.
